app/controllers/spree/admin/base_controller.rb

```diff
@@ -19,15 +19,18 @@ class BaseController < ApplicationController
 
       before_action :authorize_admin
       before_action :set_locale
-      before_action :warn_invalid_order_cycles, if: :html_request?
+      before_action :warn_invalid_order_cycles, if: :page_load_request?
 
       # Warn the user when they have an active order cycle with hubs that are not ready
       # for checkout (ie. does not have valid shipping and payment methods).
       def warn_invalid_order_cycles
-        return if flash[:notice].present?
```

Comment:
I think we should still keep this condition. It ensure we don't overwrite a more specific notice.

Reply:
change requested.

Reply:
I've addressed this change, however, I tested the scenario where there's a specific flash notice in controller action (which is the case in our app right now), then rails automatically display that and doesn't get overridden by this method because its being called before calling any action (_in before_action callback_). And every other notices are set in controller actions. So, that controller message would get the priority over this.

In future we may have a case where we would want the same scenario (setting message in before_action), then I think our base controller message would get the priority.

This check was present here because before `flash[:notice]` was being used, not it's `flash.now[:notice]`. This notice was not discarded after being displayed before and we still had that notice for the very next request. We saw that if we still had it, then no need to recalculate the notice.

Just in case I'm missing something here, pushing the requested change.
https://github.com/openfoodfoundation/openfoodnetwork/pull/12778/commits/46e54f48c91dd8863dec1bacfb0f9664f94ce32d
Thanks

Reply:
Ah ha, thanks for explaining. So this case should never occur.
Oh well, it doesn't hurt to be defensive here.